dbgpt/storage/knowledge_graph/community_summary.py

```diff
@@ -132,20 +134,95 @@ def community_store_configure(name: str, cfg: VectorStoreConfig):
     def get_config(self) -> BuiltinKnowledgeGraphConfig:
         """Get the knowledge graph config."""
         return self._config
+    
+    def _parse_chunks(slef, chunks: List[Chunk]):
+        data = []
+        for chunk_index, chunk in enumerate(chunks):
+            parent = None
+            directory_keys = list(chunk.metadata.keys())[:-1]
+            parent_level = directory_keys[-2] if len(directory_keys) > 1 else None
+            self_level = directory_keys[-1] if directory_keys else 'Header0'
+            
+            obj = {
+                'id': chunk.chunk_id,
+                'title': chunk.metadata.get(self_level, 'none_header_chunk'),
+                'directory_keys': directory_keys,
+                'level': self_level,
+                'content': chunk.content,
+                'parent_id': None,
+                'parent_title':None,
+                'type':'chunk'
+            }
+            
+            if parent_level:
+                for parent_direct in directory_keys[:-1][::-1]:
+                    parent_name = chunk.metadata.get(parent_direct, None)
+                    for n in range(chunk_index-1, -1, -1):
+                        metadata = chunks[n].metadata
+                        keys = list(metadata.keys())[:-1]
+                        if metadata and parent_direct == keys[-1] and parent_name == metadata.get(parent_direct):
+                            parent = chunks[n]
+                            obj['parent_id'] = parent.chunk_id
+                            obj['parent_title'] = parent_name
+                            break
+                        if chunk_index - n > len(directory_keys):
+                            break
+                    if obj['parent_id']:
+                        break
+            
+            if not obj['parent_id']:
+                obj['parent_id'] = 'document'
+            
+            data.append(obj)
+        
+        return data
+
+        
 
+        
+    
     async def aload_document(self, chunks: List[Chunk]) -> List[str]:
         """Extract and persist graph."""
-        # todo add doc node
-        for chunk in chunks:
-            # todo add chunk node
-            # todo add relation doc-chunk
+        
+        # check document
+        file_path = chunks[0].metadata['source'] or 'Text_Node'
+        doc_name = os.path.basename(file_path)
+        hash_id = str(uuid.uuid4())
+       
+        data_list = self._parse_chunks(chunks)
+        total_graph = MemoryGraph()
+        
+        for data in data_list:
+            chunk_src = Vertex(f"""{data['parent_id']}""",name=data["parent_title"],vertex_type=data["type"],content=data["content"])
+            chunk_dst = Vertex(f"""{data["id"]}""",name=data["title"],vertex_type=data["type"],content=data["content"])
+            chunk_include_chunk = Edge(chunk_src.vid,chunk_dst.vid,name=f"include",edge_type="chunk_include_chunk")
```

Comment:
The `edge_type` should not be directly specified at the outer layer. Let `GraphStore` decide on its own.